### PR TITLE
update sentry-native to 0.2.2

### PR DESCRIPTION
**Pull request by Swatinem**
## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

Update the `sentry-native` submodule to version `0.2.1`, which has important fixes for the module finder (fallback for libs without build-id) and a lot of other fixes overall.

## :green_heart: How did you test it?

ran the example

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing

### CODE REVIEW

File: sentry-android-ndk/build.gradle.kts [hunk 1]
```diff
@@ -39,6 +39,8 @@ android {
             cmake {
                 arguments.add(0, "-DANDROID_STL=c++_static")
                 arguments.add(0, "-DSENTRY_NATIVE_SRC=$sentryNativeSrc")
+                arguments.add(0, "-DSENTRY_BUILD_TESTS=OFF")
```
Comment (bruno-garcia) [inline sentry-android-ndk/build.gradle.kts]: We don't want tests to run anymore?

File: sentry-sample/build.gradle.kts [hunk 2]
```diff
@@ -25,6 +25,8 @@ android {
             cmake {
                 arguments.add(0, "-DANDROID_STL=c++_static")
                 arguments.add(0, "-DSENTRY_NATIVE_SRC=$sentryNativeSrc")
+                arguments.add(0, "-DSENTRY_BUILD_TESTS=OFF")
```
Comment (bruno-garcia) [inline sentry-sample/build.gradle.kts]: We don't want tests to run anymore?

File: sentry-android-ndk/build.gradle.kts (hunk 1 above)
Comment (marandaneto) [reply sentry-android-ndk/build.gradle.kts]: they never did afaik

Comment (bruno-garcia) [reply sentry-android-ndk/build.gradle.kts]: 👍

**Pull request merged**